// adaptivecards-fabric/lib/actions/ActionButton.js
"use strict";
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var office_ui_fabric_react_1 = require("office-ui-fabric-react");
var AC = require("adaptivecards");
exports.ActionButton = function (props) { return (props.iconUrl ?
    React.createElement(office_ui_fabric_react_1.CompoundButton, { className: props.className },
        React.createElement("div", { style: {
                display: "flex",
                flexDirection: props.iconPlacement === AC.ActionIconPlacement.LeftOfTitle ? "row" : "column",
                justifyContent: "center",
            } },
            React.createElement("img", { src: props.iconUrl, style: {
                    alignSelf: "center",
                    width: props.iconSize,
                    height: props.iconSize,
                    flex: "0 0 auto",
                } }),
            React.createElement("span", { style: { alignSelf: "center" } }, props.text))) :
    React.createElement(office_ui_fabric_react_1.PrimaryButton, { className: props.className, text: props.text })); };
