// adaptivecards-fabric/lib/utils/shared.js
"use strict";
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var ReactDOM = require("react-dom");
var AC = require("adaptivecards");
var office_ui_fabric_react_1 = require("office-ui-fabric-react");
exports.getDiv = function () { return document.createElement("div"); };
var sharedInternalRender = function (renderReact) {
    var div = exports.getDiv();
    ReactDOM.render(renderReact(), div);
    return div;
};
var ReactCardElement = /** @class */ (function (_super) {
    __extends(ReactCardElement, _super);
    function ReactCardElement() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ReactCardElement.prototype.internalRender = function () {
        return sharedInternalRender(this.renderReact);
    };
    return ReactCardElement;
}(AC.CardElement));
exports.ReactCardElement = ReactCardElement;
var ReactCardElementContainer = /** @class */ (function (_super) {
    __extends(ReactCardElementContainer, _super);
    function ReactCardElementContainer() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ReactCardElementContainer.prototype.internalRender = function () {
        return sharedInternalRender(this.renderReact);
    };
    return ReactCardElementContainer;
}(AC.Container));
exports.ReactCardElementContainer = ReactCardElementContainer;
var inlineButtonRootStyle = {
    marginLeft: 8,
};
var ReactInputElement = /** @class */ (function (_super) {
    __extends(ReactInputElement, _super);
    function ReactInputElement() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.handleChange = function (event, newValue) {
            _this._value = newValue;
        };
        _this.buildInlineActionButton = function () {
            return sharedInternalRender(AC.isNullOrEmpty(_this.inlineAction.iconUrl) ?
                _this.buildTextOnlyInlineActionActionButton :
                _this.inlineIconActionButton);
        };
        _this.inlineActionClickHandler = function (e) {
            e.stopPropagation();
            e.preventDefault();
            _this._inlineAction.execute();
        };
        _this.inlineIconActionButton = function () { return (React.createElement(office_ui_fabric_react_1.IconButton, { default: true, text: _this._inlineAction.title, className: _this.hostConfig.makeCssClassName("ac-inlineActionButton", "iconOnly"), styles: {
                icon: {
                    height: "100%",
                },
                root: inlineButtonRootStyle,
            }, iconProps: {
                iconType: office_ui_fabric_react_1.IconType.image,
                imageProps: {
                    height: "100%",
                    src: _this._inlineAction.iconUrl,
                },
            }, onClick: _this.inlineActionClickHandler })); };
        _this.buildTextOnlyInlineActionActionButton = function () { return (React.createElement(office_ui_fabric_react_1.PrimaryButton, { className: _this.hostConfig.makeCssClassName("ac-inlineActionButton", "textOnly"), text: _this._inlineAction.title, onClick: _this.inlineActionClickHandler, styles: {
                root: inlineButtonRootStyle,
            } })); };
        return _this;
    }
    ReactInputElement.prototype.internalRender = function () {
        var element = sharedInternalRender(this.renderReact);
        element.style.width = "100%";
        return element;
    };
    ReactInputElement.prototype.overrideInternalRender = function () {
        var inputControl = _super.prototype.overrideInternalRender.call(this);
        if (this._inlineAction) {
            this.inputControlContainerElement.appendChild(this.buildInlineActionButton());
        }
        return inputControl;
    };
    Object.defineProperty(ReactInputElement.prototype, "value", {
        get: function () {
            return this._value;
        },
        set: function (value) {
            this._value = value;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ReactInputElement.prototype, "inlineAction", {
        get: function () {
            return this._inlineAction;
        },
        set: function (action) {
            this._inlineAction = action;
            if (this._inlineAction) {
                this._inlineAction.setParent(this);
            }
        },
        enumerable: true,
        configurable: true
    });
    return ReactInputElement;
}(AC.Input));
exports.ReactInputElement = ReactInputElement;
exports.raiseParseError = function (error, errors) {
    if (errors) {
        errors.push(error);
    }
    if (AC.AdaptiveCard.onParseError != null) {
        AC.AdaptiveCard.onParseError(error);
    }
};
exports.getIntValue = function (val, errors, defaultValue) {
    if (defaultValue === void 0) { defaultValue = undefined; }
    try {
        return val ? parseInt(val, 10) : defaultValue;
    }
    catch (error) {
        exports.raiseParseError({
            error: AC.ValidationError.InvalidPropertyValue,
            message: error,
        }, errors);
    }
};
